### ISSUE TITLE
Yarn test fails

### ISSUE THREAD

**Issue by stavros-liaskos** [devCrossNet/vue-starter#270]
**Describe the bug**
`yarn test` fails

**To Reproduce**
1. `git clone https://github.com/devCrossNet/vue-starter`
2. `yarn`
3. `yarn test`

**Expected behavior**
Tests must pass:
```shell
Test Suites: 65 passed, 65 total
Tests:       228 passed, 228 total
Snapshots:   0 total
Time:        12.425s
```

**Info**
 ```shell
$ yarn -v
1.10.1
$ node -v
v10.11.0
```

**Comment by stavros-liaskos** [comment]
Pasting only the errors:

```shell
Summary of all failing tests
 FAIL  app/shared/plugins/vuex-persist/vuex-persist.spec.ts
  ● vuex-persist › should merge state

    expect(received).toEqual(expected)

    Expected value to equal:
      {"foo": {"bar": "baz"}, "initial": ["TEST"]}
    Received:
      {"initial": ["TEST"]}

    Difference:

    - Expected
    + Received

      Object {
    -   "foo": Object {
    -     "bar": "baz",
    -   },
        "initial": Array [
          "TEST",
        ],
      }

      43 |     updateHandler({ update: true }, { bar: { test: 1 } });
      44 |
    > 45 |     expect(mergedState).toEqual(
         |                         ^
      46 |       {
      47 |         foo:     {
      48 |           bar: 'baz',

      at Object.<anonymous> (app/shared/plugins/vuex-persist/vuex-persist.spec.ts:45:25)

  ● vuex-persist › should merge arrays correctly

    TypeError: Cannot read property 'calls' of undefined

      179 |                                 });
      180 |
    > 181 |     expect((window as any).localStorage.setItem.mock.calls[1])
          |                                                      ^
      182 |     .toEqual(
      183 |       [
      184 |         'vuexpersistinitial',

      at Object.<anonymous> (app/shared/plugins/vuex-persist/vuex-persist.spec.ts:181:54)

 FAIL  app/shared/plugins/vuex-persist/PersistLocalStorage.spec.ts
  ● PersistLocalStorage › should clear store

    expect(jest.fn())[.not].toHaveBeenCalled()

    jest.fn() value must be a mock function or spy.
    Received:
      function: [Function clear]

      19 |     storage.clear();
      20 |
    > 21 |     expect(window.localStorage.clear).toHaveBeenCalled();
         |                                       ^
      22 |   });
      23 |
      24 |   test('should get item', () => {

      at Object.<anonymous> (app/shared/plugins/vuex-persist/PersistLocalStorage.spec.ts:21:39)

  ● PersistLocalStorage › should get item

    expect(jest.fn())[.not].toHaveBeenCalledWith()

    jest.fn() value must be a mock function or spy.
    Received:
      function: [Function getItem]

      25 |     storage.getItem('foo');
      26 |
    > 27 |     expect(window.localStorage.getItem).toHaveBeenCalledWith('vuexpersistfoo');
         |                                         ^
      28 |   });
      29 |
      30 |   test('should get key', () => {

      at Object.<anonymous> (app/shared/plugins/vuex-persist/PersistLocalStorage.spec.ts:27:41)

  ● PersistLocalStorage › should get key

    expect(jest.fn())[.not].toHaveBeenCalledWith()

    jest.fn() value must be a mock function or spy.
    Received:
      function: [Function key]

      31 |     storage.key(1);
      32 |
    > 33 |     expect(window.localStorage.key).toHaveBeenCalledWith(1);
         |                                     ^
      34 |   });
      35 |
      36 |   test('should remove item', () => {

      at Object.<anonymous> (app/shared/plugins/vuex-persist/PersistLocalStorage.spec.ts:33:37)

  ● PersistLocalStorage › should remove item

    expect(jest.fn())[.not].toHaveBeenCalledWith()

    jest.fn() value must be a mock function or spy.
    Received:
      function: [Function removeItem]

      37 |     storage.removeItem('foo');
      38 |
    > 39 |     expect(window.localStorage.removeItem).toHaveBeenCalledWith('vuexpersistfoo');
         |                                            ^
      40 |   });
      41 |
      42 |   test('should set item and change prefix', () => {

      at Object.<anonymous> (app/shared/plugins/vuex-persist/PersistLocalStorage.spec.ts:39:44)

  ● PersistLocalStorage › should set item and change prefix

    expect(jest.fn())[.not].toHaveBeenCalledWith()

    jest.fn() value must be a mock function or spy.
    Received:
      function: [Function setItem]

      45 |     localStore.setItem('foo', 'bar');
      46 |
    > 47 |     expect(window.localStorage.setItem).toHaveBeenCalledWith('testprefixfoo', 'bar');
         |                                         ^
      48 |   });
      49 |
      50 |   test('should manipulate state before persist', () => {

      at Object.<anonymous> (app/shared/plugins/vuex-persist/PersistLocalStorage.spec.ts:47:41)


Test Suites: 2 failed, 63 passed, 65 total
Tests:       7 failed, 221 passed, 228 total
Snapshots:   0 total
Time:        18.6s
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by devCrossNet** [comment]
I get the same error with npm if it does install the latest version of jsdom. It should use version 11 of jsdom

**Comment by devCrossNet** [comment]
I would recommend to use npm then. I can not assist with yarn because I don’t use it 😕

**Comment by stavros-liaskos** [comment]
It would be nice instead of a bug to consider this a new feature. Yarn is quite faster and has other benefits too

**Comment by devCrossNet** [comment]
@stavros-liaskos I figured it out. it is not a yarn issue itself but I fixed the jsdom-localStorage issue, will be fixed in the v1.4.0 update
